src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -927,3 +928,236 @@ func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_addAndDeleteFile() {
+					isPackageOrChildOfPackage: false,
+					loc:                       path.Elements{},
+				},
+			},
+		},
+		{
+			name:      "folder hierarchy, no previous",
+			tree:      treeWithFileInFolder,
+			expectErr: require.NoError,
+			expect: map[string]collectable{
+				rootID: {
+					currPath:                  fullPathPath(t),
+					files:                     map[string]fileyMcFileFace{},
+					folderID:                  rootID,
+					isPackageOrChildOfPackage: false,
+					loc:                       path.Elements{},
+				},
+				idx(folder, "parent"): {
+					currPath:                  fullPathPath(t, namex(folder, "parent")),
+					files:                     map[string]fileyMcFileFace{},
+					folderID:                  idx(folder, "parent"),
+					isPackageOrChildOfPackage: false,
+					loc:                       path.Elements{rootName},
+				},
+				id(folder): {
+					currPath: fullPathPath(t, namex(folder, "parent"), name(folder)),
+					files: map[string]fileyMcFileFace{
+						id(file): {},
+					},
+					folderID:                  id(folder),
+					isPackageOrChildOfPackage: false,
+					loc:                       path.Elements{rootName, namex(folder, "parent")},
+				},
+			},
+		},
+		{
+			name: "package in hierarchy",
+			tree: func(t *testing.T) *folderyMcFolderFace {
+				ctx, flush := tester.NewContext(t)
+				defer flush()
+
+				tree := treeWithRoot(t)
+				err := tree.setFolder(ctx, rootID, id(pkg), name(pkg), true)
+				require.NoError(t, err, clues.ToCore(err))
+
+				err = tree.setFolder(ctx, id(pkg), id(folder), name(folder), false)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return tree
+			},
+			expectErr: require.NoError,
+			expect: map[string]collectable{
+				rootID: {
+					currPath:                  fullPathPath(t),
+					files:                     map[string]fileyMcFileFace{},
+					folderID:                  rootID,
+					isPackageOrChildOfPackage: false,
+					loc:                       path.Elements{},
+				},
+				id(pkg): {
+					currPath:                  fullPathPath(t, name(pkg)),
+					files:                     map[string]fileyMcFileFace{},
+					folderID:                  id(pkg),
+					isPackageOrChildOfPackage: true,
+					loc:                       path.Elements{rootName},
+				},
+				id(folder): {
+					currPath:                  fullPathPath(t, name(pkg), name(folder)),
+					files:                     map[string]fileyMcFileFace{},
+					folderID:                  id(folder),
+					isPackageOrChildOfPackage: true,
+					loc:                       path.Elements{rootName, name(pkg)},
+				},
+			},
+		},
+		{
+			name:      "folder hierarchy with previous paths",
+			tree:      treeWithFileInFolder,
+			expectErr: require.NoError,
+			prevPaths: map[string]string{
+				rootID:                fullPath(),
+				idx(folder, "parent"): fullPath(namex(folder, "parent-prev")),
+				id(folder):            fullPath(namex(folder, "parent-prev"), name(folder)),
```

Comment:
nice! I like that the old and new paths are different so we can tell if we're accidentally reusing one of them somehow